service/src/main/java/uk/gov/hmcts/reform/fpl/service/PastHearingDatesValidatorService.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class PastHearingDatesValidatorService {
+    private static final LocalTime MIDNIGHT = LocalTime.of(0, 0, 0);
+
+    public List<String> validateHearingDates(LocalDateTime hearingStartDate, LocalDateTime hearingEndDate) {
+        List<String> errors = new ArrayList<>();
+
+        if (hearingStartDate != null && !isValid(hearingStartDate)) {
```

Comment:
This is confusing, double negating here on private method. What about

```
@Service
@RequiredArgsConstructor(onConstructor = @__(@Autowired))
public class PastHearingDatesValidatorService {
    private static final LocalTime MIDNIGHT = LocalTime.of(0, 0, 0);
    public List<String> validateHearingDates(LocalDateTime hearingStartDate, LocalDateTime hearingEndDate) {
        List<String> errors = new ArrayList<>();
        if (hearingStartDate != null && isMidnight(hearingStartDate)) {
            errors.add("Enter a valid start time");
        }
        if (hearingEndDate != null && isMidnight(hearingEndDate)) {
            errors.add("Enter a valid end time");
        }
        return errors;
    }
    private boolean isMidnight(LocalDateTime localDateTime) {
        return localDateTime.toLocalTime().equals(MIDNIGHT);
    }
}
```

Reply:
yup this is nicer :)